src/cls/IPM/Repo/Oras/Definition.cls

```diff
@@ -0,0 +1,183 @@
+Class %IPM.Repo.Oras.Definition Extends %IPM.Repo.Definition
+{
+
+Parameter DISPLAYNAME As STRING = "ORAS";
+
+Parameter MONIKER As STRING = "oras";
+
+Property URL As %String(MAXLEN = 2048) [ Required ];
+
+Index ORASURL On URL [ Unique ];
+
+Parameter URLPromptString = {$$$Text("ORAS Registry URL:","ZPM")};
+
+Property Namespace As %String;
+
+Property Username As %String(MAXLEN = "");
+
+Property Password As %String(MAXLEN = "");
+
+Property Token As %String(MAXLEN = "");
+
+Property DeploymentEnabled As %Boolean [ InitialExpression = 0 ];
+
+Index DeploymentServerOras On DeploymentEnabled [ Unique ];
+
+Property Details As %String(MAXLEN = "") [ SqlComputeCode = {Set {*} = {URL}}, SqlComputed, SqlComputeOnChange = (%%INSERT, %%UPDATE) ];
+
+XData Commands [ XMLNamespace = "http://www.intersystems.com/PackageManager/CLI" ]
+{
+<commands>
+<command name="repo">
+<group name="Oras repository">
+
+<modifier name="namespace" aliases="ns" value="true" description="For ORAS repositories, specifies the registry namespace, i.e. the fully specified path to a package is &lt;url>/&lt;namespace>/&lt;package>.
+	Note, this is not the IRIS namespace, but a namespace in the sense of a (sub)directory in the registry's storage system. For example, if in a custom managed registry, the base url is 'registry.company.com',
+	and the package is 'objectscript-example', but it will be stored at 'registry.company.com/packages/ipm/objectscript-example', then the namespace will be 'packages/ipm'." />
+</group>
+</command>
+</commands>
+}
+
+/// This callback method is invoked by the <METHOD>%Save</METHOD> method to 
+/// provide notification that the object is being saved. It is called before 
+/// any data is written to disk.
+/// 
+/// <P><VAR>insert</VAR> will be set to 1 if this object is being saved for the first time.
+/// 
+/// <P>If this method returns an error then the call to <METHOD>%Save</METHOD> will fail.
+Method %OnBeforeSave(insert As %Boolean) As %Status [ Private, ServerOnly = 1 ]
+{
+	// Leave DeploymentEnabled null to make unique index work right
+	// (i.e., deployment can only be enabled for one remote server.)
```

Comment:
Why can deployment be enabled only for one remote server? If it is enabled for multiple, I think the user should be able to choose which one to deploy to at time of publishing

Reply:
The field is misleadingly named (see [514](https://github.com/intersystems/ipm/issues/514)) and actually is specifying the default. The repo to be published to can be specified using the "-r" flag.